src/sentry/incidents/endpoints/project_alert_rule_index.py

```diff
@@ -8,11 +10,122 @@
 
+class CombinedRuleSerializer(Serializer):
+    def serialize(self, obj, attrs, user, **kwargs):
+        rule = obj
+        # TODO: Use native serializer instead of cut/paste
+        if isinstance(rule, AlertRule):
+            # serializer = RuleSerializer()
+            return {
+                "type": "alert-rule",
+                "id": six.text_type(rule.id),
+                "name": rule.name,
+                "organizationId": six.text_type(rule.organization_id),
+                "status": rule.status,
+                # TODO: Remove when frontend isn't using
+                "thresholdType": 0,
+                "dataset": rule.dataset,
+                "query": rule.query,
+                "aggregation": rule.aggregation,
+                "aggregations": [rule.aggregation],
+                "timeWindow": rule.time_window,
+                "resolution": rule.resolution,
+                # TODO: Remove when frontend isn't using
+                "alertThreshold": 0,
+                # TODO: Remove when frontend isn't using
+                "resolveThreshold": 0,
+                "thresholdPeriod": rule.threshold_period,
+                "triggers": attrs.get("triggers", []),
+                "includeAllProjects": rule.include_all_projects,
+                "dateModified": rule.date_modified,
+                "dateAdded": rule.date_added,
+            }
+        elif isinstance(rule, Rule):
+            # serializer = AlertRuleSerializer()
+            environment = None  # attrs["environment"]
+            return {
+                "type": "rule",
+                "id": six.text_type(rule.id) if rule.id else None,
+                "conditions": [
+                    dict(o.items() + [("name", _generate_rule_label(rule.project, obj, o))])
+                    for o in rule.data.get("conditions", [])
+                ],
+                "actions": [
+                    dict(o.items() + [("name", _generate_rule_label(rule.project, obj, o))])
+                    for o in rule.data.get("actions", [])
+                ],
+                "actionMatch": rule.data.get("action_match") or Rule.DEFAULT_ACTION_MATCH,
+                "frequency": rule.data.get("frequency") or Rule.DEFAULT_FREQUENCY,
+                "name": rule.label,
+                "dateCreated": rule.date_added,
+                "environment": environment.name if environment is not None else None,
+            }
+        else:
+            raise AssertionError("Invalid rule to serialize: %r" % type(rule))
+
+
+class ProjectCombinedRuleIndexEndpoint(ProjectEndpoint):
+    def get(self, request, project):
+        """
+        Fetches alert rules and legacy rules for an organization
+        """
+        if not features.has("organizations:incidents", project.organization, actor=request.user):
+            raise ResourceDoesNotExist
+
+        # TODO: How to implement cursor in results?
+        cursor = 0  # TODO: Get from request. Is it an integer? Date?
```

Comment:
One other tricky thing (that maybe is fine): If two rows from different tables have the exact same creation date, and the page boundary is between them, then we need to use offset to know which to start on